explorer/idl/web3api.idl

```diff
@@ -0,0 +1,225 @@
+    gasUsed             int
+    cumulativeGasUsed   int
+    contractAddress     string
+    logs                []Log
+    logsBloom           string
+}
+
+struct Block {
+    number              int
+    hash                string
+    parentHash          string
+    nonce               string
+    sha3Uncles          string
+    logsBloom           string
+    transactionsRoot    string
+    stateRoot           string
+    receiptsRoot        string
+    miner               string
+    difficulty          int
+    totalDifficulty     int
+    extraData           string
+    size                int
+    gasLimit            int
+    gasUsed             int
+    timestamp           int
+    transactions        []Transaction
+    uncles              []string
+}
+
+interface Web3API {
+    // ClientVersion returns the current client version
+    ClientVersion() string
+
+    // Sha3 returns Keccak-256 (not the standardized SHA3-256) of the given data
+    Sha3(input string) string
+
+    // NetVersion returns the current net id
+    NetVersion() string
+
+    // NetListening returns whether client is actively listening for network connections
+    NetListening() bool
+
+    // NetPeerCount returns number of peers currently connected to the client
+    NetPeerCount() int
+
+    // ProtocolVersion returns the current iotex protocol version
+    ProtocolVersion() string
+
+    // Syncing returns an object with data about the sync status or false
+    Syncing() string
+
+    // Coinbase returns the client coinbase address
+    Coinbase() string
+
+    // Mining returns true if client is actively mining new blocks
+    Mining() string
+
+    // HashRate returns the number of hashes per second that the node is mining with
+    HashRate() string
+
+    // GasPrice returns the current price per gas
+    GasPrice() int
+
+    // Accounts returns a list of addresses owned by client
+    Accounts() []string
+
+    // BlockNumber returns the number of most recent block
+    BlockNumber() int
+
+    // GetBalance returns the balance of the account of given address
+    // position can be a block number or 'latest', 'earliest' and 'pending'
+    GetBalance(address string, blockNumber int) int
+
+    // GetStorageAt returns the value from a storage position at a given address
+    GetStorageAt(address string, key int, blockNumber int) string
+
+    // GetTransactionCount returns the number of transactions sent from an address
+    GetTransactionCount(address string, blockNumber int) int
+
+    // GetBlockTransactionCountByHash returns the number of transactions in a block from a
+    // block matching the given block hash
+    GetBlockTransactionCountByHash(blockHash string) int
+
+    // GetBlockTransactionCountByNumber returns the number of transactions in a block matching
+    // the given block number
+    GetBlockTransactionCountByNumber(blockNumber int) int
+
+    // GetUncleCountByBlockHash returns the number of uncles in a block from a block matching
```

Comment:
what about uncle?

Reply:
guess it means "uncle block" -- which should not exist in iotex root chain

Reply:
yeah, we don't have the uncle concept, I will keep the interface and return un supported, so developer can know.